fuel-tx/src/transaction/id.rs

```diff
@@ -682,6 +686,15 @@ mod tests {
         let scripts = vec![vec![], generate_bytes(rng), generate_bytes(rng)];
         let script_data = vec![vec![], generate_bytes(rng), generate_bytes(rng)];
         let storage_slots = vec![vec![], vec![rng.gen(), rng.gen()]];
+        let purposes = [
```

Comment:
This test is giant. We should probably split each `Vec` into its own little helper...? 


We should also probably come up with a better name for this test, `id` isn't really a behavior. `id__changing_fields_in_the_tx_changes_the_generated_id`? or something like that?

_You don't need to do this, but we wouldn't need all the nested `for` loops if we just used prop test strategies to generate the groups. But that would be kinda a lot of work to set up, so I think this is fine for now_

Reply:
https://github.com/FuelLabs/fuel-vm/issues/717